### PR TITLE
Update php doc to reflect proper return type

### PR DESCRIPTION
**Pull request by tomneedham**
## Description
Otherwise PHPStan complains when you supply `null` as the default and do a comparison of the output like this:

```PHP
$out = $config->getAppValue('myapp', 'mykey', null);
if($out === null) {
// do something
}
```

## Motivation and Context
PHPStan failure 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Review by DeepDiver1975 — APPROVED** [review]
Just had the same reported ;-)

File: lib/public/IConfig.php [hunk 1]
```diff
@@ -117,7 +117,7 @@ public function setAppValue($appName, $key, $value);
 	 * @param string $appName the appName that we stored the value under
 	 * @param string $key the key of the value, under which it was saved
 	 * @param string $default the default value to be returned if the value isn't set
```
Comment (DeepDiver1975) [inline lib/public/IConfig.php]: should be mixed as well - right?

**Review by patrickjahns — APPROVED** [review]
😍  - thanks guys

**Comment by phil-davis** [comment]
Backport ``stable10`` #32427

**Pull request merged**